File: 2-sinif/4-yariyil/pdp/hft07/java_eclipse/hft07/src/hft07/Kisi.java
package hft07;

public class Kisi {
	private String isim;
	private int yas;
	private float boy;
	private float kilo;

	public Kisi(String isim, float boy, float kilo) {
		this.isim = isim;
		this.boy = boy;
		this.kilo = kilo;
		this.yas = 0;
	}

	private void sindirimSistemi(float kalori) {
		this.kilo += (kalori / 1000);
		if (this.yas < 20)
			this.boy++;
	}

	public void yemekYe(float kalori) {
		sindirimSistemi(kalori);
	}

	public void yaslan(int yil) {
		this.yas += yil;
	}

	public String getIsim() {
		return isim;
	}

	public float getKilo() {
		return kilo;
	}

}

File: 2-sinif/4-yariyil/pdp/hft07/c_vscode/NesneYonelimliBenzetim/src/Kisi.c
#include "Kisi.h"

void sindirimSistemi(const Kisi this, float kalori) { //private fonksiyon
		this->kilo += (kalori/1000);
		if(this->yas < 20) this->boy++;
}

Kisi new_Kisi(char* isim, float boy, float kilo){
	Kisi this;
	this = (Kisi)malloc(sizeof(struct KISI)); //buraya KISI* yazmıyoruz KISI'nin kapladığı yer kadar
	this->isim = isim;
	this->boy = boy;
	this->kilo = kilo;
	this->yas = 0;
	return this; //Javada otomatik isim döndürür biz elle döndürücez
}
void yemekYe(const Kisi this, float kalori){
	sindirimSistemi(this, kalori);
}
void yaslan(const Kisi this, int yil){
	this->yas += yil;
}
char* getIsim(const Kisi this){
	return this->isim;
}
float getKilo(const Kisi this){
	return this->kilo;
}
void delete_Kisi(const Kisi this){
	free(this);
}


